scripts/devenv.py

```diff
@@ -1,272 +1,94 @@
-    add_global_args(remove_ap)
+    logging.info("command %r started", args.cmd)
+    time_start = time.perf_counter()
 
-    args = ap.parse_args()
+    try:
+        args.func(args)
+    except subprocess.CalledProcessError:
+        logging.error("command %r failed", args.cmd)
+    except KeyboardInterrupt:
+        logging.error("command %r interrupted", args.cmd)
+    else:
+        logging.info("command %r completed", args.cmd)
 
-    if shutil.which("conda") is None:
-        logging.error("conda is not installed")
-        return
+    time_end = time.perf_counter()
+    logging.info("elapsed time: %.3f seconds", time_end - time_start)
 
-    if run_returncode("conda", "build", "--version") != 0:
-        logging.error("conda-build is not installed; run 'conda install -n base -c conda-forge conda-build' to install")
-        return
 
-    env = CondaEnv(args.name)
-    execution_time, _res = timed(args.func, args, env)
-    logging.info(f"finished {args.command} ({execution_time:.3f} seconds)")
+def action_create(args):
+    run(["conda", "install", "--yes", "--name", "base", "--channel", "conda-forge", "conda-build"])
 
+    chans = ["--channel", "ilastik-forge", "--channel", "conda-forge"]
+    pkgs = ["ilastik-dependencies-no-solvers", "pre-commit"]
+    run(["conda", "create", "--yes", "--name", args.name] + chans + pkgs)
 
-def command_create(args: argparse.Namespace, env: CondaEnv) -> None:
-    if env.exists():
-        logging.error(f"environment {env.name} already exists")
-        return
+    for repo in REPOSITORIES:
+        run(["conda", "develop", "--name", args.name, repo])
+        run(["conda", "run", "--name", args.name, "pre-commit", "install"], cwd=repo)
 
-    location = args.location
-    if location is not None:
-        location = pathlib.Path(location).resolve()
-        if not location.is_dir():
-            logging.error(f"{location} is not a valid directory")
-            return
 
-    logging.info(
-        f"create environment {env.name}"
-        f" with packages [{', '.join(args.package)}]"
-        f" from channels [{', '.join(args.channel)}]"
-    )
-    env.create(packages=args.package, channels=args.channel)
+def action_remove(args):
+    run(["conda", "env", "remove", "--yes", "--name", args.name])
 
-    if location is None:
-        return
 
-    packages = "volumina", "ilastik"
+def run(cmd, cwd=None):
+    msg = "run command " + repr(shlex.join(cmd))
+    if cwd:
+        msg += " in working directory " + repr(cwd)
+    logging.info(msg)
+    subprocess.run(cmd, stdout=sys.stdout, stderr=subprocess.STDOUT, cwd=cwd, check=True)
 
-    logging.info(f"Installing local ilastik packages in development mode.")
-    for pkg in packages:
-        env.install_pacakge_dev(location / pkg)
 
+def is_git_repo(cwd):
+    if not os.path.isdir(cwd):
+        return False
+    proc = subprocess.run(["git", "rev-parse"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, cwd=cwd)
+    return proc.returncode == 0
```

Comment:
I see you're adopting the unified `run` from subprocess.

subprocess docs [state](https://docs.python.org/3/library/subprocess.html#subprocess.check_call)
> Code needing to capture stdout or stderr should use run() instead:

so not sure here.

Reply:
> Do I see correctly, that it is no longer possible to override channels, add custom packages? I for one have used it a lot.

I wanted to simplify this script to the point that you could only use it to automate steps from `CONTRIBUTING.md`. My intention was that if someone wants to have other stuff installed, they could do themselves later in the environment.

But if you were using this a lot, I'll add this back.

Reply:
In this specific instance `check_call` is good because we don't need to capture.